packages/sign_in_with_apple/lib/sign_in_with_apple.dart

```diff
@@ -21,35 +32,137 @@ export './src/widgets/sign_in_with_apple_button.dart'
 // ignore: avoid_classes_with_only_static_members
 class SignInWithApple {
   @visibleForTesting
-  static const channel =
-      MethodChannel('de.aboutyou.mobile.app.sign_in_with_apple');
+  static const channel = MethodChannel(
+    'com.aboutyou.dart_packages.sign_in_with_apple',
+  );
+
+  /// Returns the credentials stored in the Keychain for the app-associated websites
+  static Future<AuthorizationCredentialPassword> getKeychainCredential() async {
+    try {
+      return parseAuthorizationCredentialPassword(
+        await channel.invokeMethod<Map<dynamic, dynamic>>(
+          'performAuthorizationRequest',
+          [
+            PasswordAuthorizationRequest(),
+          ].map((request) => request.toJson()).toList(),
+        ),
+      );
+    } on PlatformException catch (exception) {
+      throw SignInWithAppleException.fromPlatformException(exception);
+    }
+  }
 
-  /// Request credentials from the system, preferring existing keychain credentials
-  /// over "Sign in with Apple"
+  /// Requests Apple ID credentials from the system.
   ///
-  /// When no credentials are returned (e.g. also by the user cancelling), this is treated as
-  /// all other errors (just like the native API), and will throw an Exception
+  /// Fields on the returned [AuthorizationCredential] will be set based on the given scopes,
+  /// and user data will only be set if this is the initial authentication for the Apple ID with your app.
   ///
-  /// Successful result will be either of type [AuthorizationCredentialAppleID] or [AuthorizationCredentialPassword]
-  static Future<AuthorizationCredential> requestCredentials() async {
-    return parseCredentialsResponse(
-      await channel
-          .invokeMethod<Map<dynamic, dynamic>>('performAuthorizationRequest'),
-    );
+  /// In case of an error on the native side, we will throw an [SignInWithAppleException].
+  /// If we have a more specific authorization error, we will throw [SignInWithAppleAuthorizationError],
+  /// which has more information about the failure.
+  static Future<AuthorizationCredential> getAppleIDCredential({
+    @required List<AppleIDAuthorizationScopes> scopes,
+
+    /// Optional parameters for web-based authentication flows on non-Apple platforms
+    WebAuthenticationOptions webAuthenticationOptions,
+  }) async {
+    assert(scopes != null);
+
+    if (webAuthenticationOptions == null &&
+        (!Platform.isIOS && !Platform.isMacOS)) {
+      throw Exception(
+        'webAuthenticationOptions parameter must be provided on non-Apple platforms',
+      );
+    }
+
+    if (Platform.isAndroid) {
+      return _signInWithAppleAndroid(
+        scopes: scopes,
+        webAuthenticationOptions: webAuthenticationOptions,
+      );
+    }
+
```

Comment:
Should we check here that we are on iOS or macOS? and if not we should just throw an `Unsupported` exception